src/Chainweb/BlockHeader.hs

```diff
@@ -322,6 +323,7 @@ epochStart
     -> EpochStartTime
         -- ^ epoch start time of new block
 epochStart ph@(ParentHeader p) adj (BlockCreationTime bt)
+    | Nothing <- effectiveWindow p = _blockEpochStart p
```

Comment:
This is for test-related chainweb versions yeah?

Reply:
yes